### PR TITLE
Remove the unused Surface param from convert alpha docs

### LINKED ISSUE #1753: docs: document `Surface.convert_alpha()` silently ignoring argument for now (maybe also issue warnings for arguments passed?) (3557)
#  **Issue &numero;3557 opened by *[francis-clairicia](https://github.com/francis-clairicia)* at 2022-11-12 11:12:23**

Currently, `Surface.convert_alpha()` always uses the display surface, ignoring the one as parameter.

https://github.com/pygame/pygame/blob/0164760b2a3b603fab111f2dc95d170d95783db4/src_c/surface.c# L1663

And the documentation does not mention this.

It would be nice to inform the users that giving this setting does not change anything (for now or forever ?), or something else.<hr>

#  Comments

# # #  **[Matiiss](https://github.com/Matiiss)* commented at 2022-11-13 11:03:01*

It looks like they need to figure it, so I guess, it's kind of temporary until it gets fixed I suppose. Also, doesn't that warning already inform users?
<hr>

# # #  **[francis-clairicia](https://github.com/francis-clairicia)* commented at 2022-11-13 12:13:58*

Nope, the argument is just silently ignored at runtime
<hr>

# # #  **[ankith26](https://github.com/ankith26)* commented at 2022-11-14 11:17:47*

This is a known issue (duplicate of # 599).

> And the documentation does not mention this.

Good call, this should be documented for now, because this issue is tricky and may take a while to properly address. I will just re-purpose this issue as a docs issue, since the other issue is already tracking the underlying issue

### PR DESCRIPTION
**Pull request by MyreMylar**
It does nothing at all.

fixes #1753

### CODE REVIEW

**Comment by zoldalma999** [comment]
Apart from changing the docs, we could also remove the stubs for it and deprecate it in code. That would also close #493. Unless we want to add this functionality back?

**Comment by MyreMylar** [comment]
> Apart from changing the docs, we could also remove the stubs for it and deprecate it in code. That would also close #493. Unless we want to add this functionality back?

Pretty sure we don't want this back.

I've added the deprecation warnings.

**Review by Temmie3754 — APPROVED** [review]
No longer appears in docs, stubs and appropriate depreciation warning displays

File: docs/reST/ref/surface.rst [hunk 1]
```diff
@@ -240,17 +240,15 @@
...
+      with per pixel alpha.
 
       Unlike the :meth:`convert()` method, the pixel format for the new
-      image will not be exactly the same as the requested source, but it will
-      be optimized for fast alpha blitting to the destination.
+      image will not be exactly the same as the display surface, but it will
+      be optimized for fast alpha blitting to it.
 
       As with :meth:`convert()` the returned surface has the same class as
       the converted surface.
```
Comment (ankith26) [inline docs/reST/ref/surface.rst:254]: A note here of the deprecation would be nice

**Review by ankith26 — CHANGES_REQUESTED** [review]
Also a small and simple test that tests the deprecation warning would also be nice

File: docs/reST/ref/surface.rst (hunk 1 above)
Comment (MyreMylar) [reply docs/reST/ref/surface.rst:254]: 👍 added a `versionchanged` note.

**Comment by MyreMylar** [comment]
> Also a small and simple test that tests the deprecation warning would also be nice

Added a test for the deprecation warning, and removed a few places in the test that were now triggering it as they were passing in the surface argument (which has done nothing for years).

File: test/surface_test.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 except (ImportError, NameError):
     pass
 import pygame
+import pygame
```
Comment (zoldalma999) [inline test/surface_test.py:14]: this seems accidental

**Review by zoldalma999 — APPROVED** [review]
Apart from a minor nitpick, this looks good to me

File: test/surface_test.py (hunk 2 above)
Comment (MyreMylar) [inline test/surface_test.py:14]:
```suggestion
```

Comment (MyreMylar) [reply test/surface_test.py]: oops, thanks 👍 

Fixed.

**Review by ankith26 — APPROVED** [review]
LGTM now, thanks for the PR :tada:

**Pull request merged**